src/p11_child/p11_child_openssl.c

```diff
@@ -582,6 +597,7 @@ errno_t init_p11_ctx(TALLOC_CTX *mem_ctx, const char *ca_db,
         return ENOMEM;
     }
 
+    ctx->start_timestamp = time(NULL);
```

Comment:
Hi,

why don't you put `timeout` into `struct p11_ctx` as well (it would fit into `struct cert_verify_opts` as well) ? This would help to avoid adding `timeout` into all the function headers.

bye,
Sumit

Reply:
I was thinking of that too, I have no preference here. But before proceeding with changes I want to do some testing to see if this solves the issue correctly. This is still draft PR.

Reply:
I would put this into `cert_verify_opts` as it would reduce number of changes (this struct is populated already in `main()`). But I don't insist.

Reply:
I prefer  to keep it in state - particularly after change to deadline.